Content.Server/Anomaly/Effects/IceAnomalySystem.cs

```diff
@@ -0,0 +1,154 @@
+/// </summary>
+public sealed class IceAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly EntityManager _entman = default!;
+    [Dependency] private readonly AtmosphereSystem _atmosphere = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly TransformSystem _xform = default!;
+    [Dependency] private readonly GunSystem _gunSystem = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ExplosionSystem _boom = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<IceAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<IceAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
+    }
+
+    private void OnPulse(EntityUid uid, IceAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        ShootProjectilesAtEntites(uid, component, args.Severity, args.Stability);
+    }
+
+    private void ShootProjectilesAtEntites(EntityUid uid, IceAnomalyComponent component, float severity, float stability)
+    {
+        var xform = Transform(uid);
+        var projectilesShot = 0;
+
+        foreach (var entity in _lookup.GetEntitiesInRange(uid, component.ProjectileRange * stability, LookupFlags.Dynamic))
+        {
+            if (projectilesShot >= component.MaxProjectiles * severity)
+                return;
+
+            // Living entities are more likely to be shot at then non living
+            if (!HasComp<MobStateComponent>(entity) && _random.Prob(0.5f))
+                continue;
+
+            var targetCoords = Transform(entity).Coordinates.Offset(_random.NextVector2(-1, 1));
+
+            ShootProjectile(
+                uid, component,
+                xform.Coordinates,
+                targetCoords,
+                severity
+            );
+            projectilesShot++;
+        }
+    }
+
+    private void ShootProjectile(
+        EntityUid uid,
+        IceAnomalyComponent component,
+        EntityCoordinates coords,
+        EntityCoordinates targetCoords,
+        float severity
+        )
+    {
+        var mapPos = coords.ToMap(_entman, _xform);
+
+        var spawnCoords = _mapManager.TryFindGridAt(mapPos, out var grid)
+                ? coords.WithEntityId(grid.Owner, EntityManager)
+                : new(_mapManager.GetMapEntityId(mapPos.MapId), mapPos.Position);
+
+        var ent = Spawn(component.ProjectilePrototype, spawnCoords);
+        var direction = targetCoords.ToMapPos(_entman, _xform) - mapPos.Position;
+
+        if (!TryComp<ProjectileComponent>(ent, out var comp))
+            return;
+
+        foreach (var key in component.ProjectileDamage.DamageDict.Keys)
+        {
+            comp.Damage.DamageDict[key] = component.ProjectileDamage.DamageDict[key] * severity;
+        }
```

Comment:
The projectile damage should really be on the projectile, not here.
That way, you can just multiply `component.ProjectileDamage` by the severity instead of doing this weird assignment thing.